include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -11,7 +11,7 @@
 #define _ONEDPL_SYCL_BACKEND_IMPL_H
 
 #include <CL/sycl.hpp>
-#include "oneapi/dpl/internal/dynamic_selection.h"
+#include "oneapi/dpl/dynamic_selection"
```

Comment:
This is a cyclic include. Needs to be addressed.  This is seems necessary because of the use of the execution_info reporting.  Reporting is part of the contract between the policies and backends, so probably does not need to be directly in the <dynamic_selection> header, but instead included by the policies and backends that opt-in to the use of execution_info contract.

Reply:
If this include the new dynamic_selection_traits.h we can break the cycle.

Reply:
Will be fixed in #1119